deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/BaseDeepLinkDelegate.java

```diff
@@ -23,25 +23,51 @@
   protected static final String TAG = "DeepLinkDelegate";
 
   protected final List<? extends BaseRegistry> registries;
+  /**
+   * <p>Mapping of values for DLD to substitute for annotation-declared configurablePathSegments.
+   * </p>
+   * <p>Example</p>
+   * Given:
+   * <ul>
+   * <li><xmp>@DeepLink("https://www.example.com/<replaceable-path-variable>/users/{param1}")
+   * </xmp></li>
+   * <li>mapOf("pathVariableReplacementValue" to "obamaOs")</li>
+   * </ul>
+   * Then:
+   * <ul><li><xmp>https://www.example.com/obamaOs/users/{param1}</xmp> will match.</li></ul>
+   */
+  protected final Map<String, String> configurablePathSegmentReplacements;
 
   public List<? extends BaseRegistry> getRegistries() {
     return registries;
   }
 
   public BaseDeepLinkDelegate(List<? extends BaseRegistry> registries) {
     this.registries = registries;
+    configurablePathSegmentReplacements = new HashMap<>();
+  }
+
+  public BaseDeepLinkDelegate(
+    List<? extends BaseRegistry> registries,
+    Map<String, String> configurablePathSegmentReplacements
+  ) {
+    this.registries = registries;
+    this.configurablePathSegmentReplacements = configurablePathSegmentReplacements;
+    ValidationUtilsKt.validateConfigurablePathSegmentReplacements(registries,
```

Comment:
I kinda don't want to do that on every start, this is something that cannot change after we build the app, isn't there a way to do this at compile time?